### PR TITLE
Router -> Controller -> Model Pattern for User 

### PR DESCRIPTION
**Pull request by AndyLiang1**
Created the Router -> Controller -> Model Pattern for User. This PR also contains code to connect to our mongoDB.

Router calls controllers and is responsible for only exposing the route. 
Controller contains business logic and accesses the DB. Controllers may also call other controllers. 
Models define the schema of a collection (think of a table in SQL). 

To test this PR, you need a .env file in the server folder. See the secrets channel. 
In postman or google, put in http://localhost:5000/api/user to see the result. 
image

### CODE REVIEW

File: server/src/config/config.js
Comment (L0Lmaker) [inline server/src/config/config.js:1]: I don't think we need a src folder, can move the items in src to `server/`

File: server/index.js [hunk 1]
```diff
@@ -1,11 +1,36 @@
...
+    await mongoose.connect(config.mongo.uri)
+    startServer()
+  } catch (err) {
+    console.error(err)
+  }
+}
+
+const startServer = () => {
+  app.listen(config.server.port, () => {
+    console.log(`Server is running on port ${config.server.port}`);
```
Comment (L0Lmaker) [inline server/index.js:32]: @NLmeng was the linting  rule not enforced over here for the semi-colon?

File: server/yarn.lock [hunk 2]
```diff
@@ -2,6 +2,25 @@
...
+  integrity sha512-xTE1E+YF4aWPJJeUzaZI5DRntlkY3+BCVJi0axFptnjGmAoWxkyREIh/XMrfxVLejwQxMCfDXdICo0VLxThrog==
+
+"@types/whatwg-url@^8.2.1":
+  version "8.2.2"
+  resolved "https://registry.yarnpkg.com/@types/whatwg-url/-/whatwg-url-8.2.2.tgz#749d5b3873e845897ada99be4448041d4cc39e63"
+  integrity sha512-FtQu10RWgn3D9U4aazdwIE2yzphmTJREDqNdODHrbrZmmMqI0vMheC/6NE/J1Yveaj8H+ela+YwWTjq5PGmuhA==
+  dependencies:
+    "@types/node" "*"
+    "@types/webidl-conversions" "*"
+=======
```
Comment (L0Lmaker) [inline server/yarn.lock]: This looks like a merge conflict

File: server/yarn.lock [hunk 3]
```diff
@@ -1470,6 +1502,12 @@ ignore@^4.0.6:
   resolved "https://registry.yarnpkg.com/ignore/-/ignore-4.0.6.tgz#750e3db5862087b4737ebac8207ffd1ef27b25fc"
   integrity sha512-cyFDKrqc/YdcWFniJhzI42+AzS+gNwmUzOSFcRCQYwySuBBBy/KjuxWLZ/FHEH6Moq1NizMOBWyTcv8O4OZIMg==
 
+<<<<<<< HEAD
+ip@^2.0.0:
+  version "2.0.0"
+  resolved "https://registry.yarnpkg.com/ip/-/ip-2.0.0.tgz#4cf4ab182fee2314c75ede1276f8c80b479936da"
+  integrity sha512-WKa+XuLG1A1R0UWhl2+1XQSi+fZWMsYKffMZTTYsiZaUD8k2yDAj5atimTUD2TZkyCkNEeYE5NhFZmupOGtjYQ==
+=======
```
Comment (L0Lmaker) [inline server/yarn.lock]: here too

File: server/yarn.lock (hunk 2 above)
Comment (L0Lmaker) [reply server/yarn.lock]: These should be resolved if you rebase onto main, make a backup of this branch just incase the rebase messes things up.

**Review by L0Lmaker — CHANGES_REQUESTED** [review]
Resolve Merge conflicts and restructure files, otherwise LGTM

**Comment by L0Lmaker** [comment]
I think you have to whitelist public IPs to connect to the Atlas cluster @AndyLiang1

**Comment by L0Lmaker** [comment]
Related to #28

**Comment by L0Lmaker** [comment]
@AndyLiang1 Can you also add instructions for local development without connecting to Atlas Cluster, incase we want to perform testing, local environment and data would be super useful so we dont nuke the "production" database

File: server/src/config/config.js
Comment (AndyLiang1) [reply server/src/config/config.js:1]: Sure

File: server/index.js (hunk 1 above)
Comment (AndyLiang1) [reply server/index.js:32]: Sorry yea, I wrote a sticky note to run the esllint stuff...I merged dev after I created this PR.

**Comment by AndyLiang1** [comment]
> @AndyLiang1 Can you also add instructions for local development without connecting to Atlas Cluster, incase we want to perform testing, local environment and data would be super useful so we dont nuke the "production" database

Not actually sure how to do local development with Atlas... I will look into this soon

**Comment by AndyLiang1** [comment]
> I think you have to whitelist public IPs to connect to the Atlas cluster @AndyLiang1

My thinking is just whoever uses it would whitelist their own IPs...I should add this to the read me

File: server/yarn.lock (hunk 2 above)
Comment (AndyLiang1) [reply server/yarn.lock]: Oh what the...hmmmmm I will need to think about this one

Comment (Graywing13) [reply server/yarn.lock]: Rithin's first comment is right. As for the second one -- you've already merged main into your PR, hence why you see the `<<<<<< HEAD`. So you just need to comb out the knots in the merge conflict by deleting / reordering where necessary.

File: server/package.json [hunk 4]
```diff
@@ -16,15 +16,9 @@
     "add": "^2.0.6",
     "cors": "^2.8.5",
     "dotenv": "^16.0.3",
-    "eslint": "^8.41.0",
-    "eslint-config-node": "^4.1.0",
-    "eslint-config-prettier": "^8.8.0",
-    "eslint-plugin-node": "^11.1.0",
-    "eslint-plugin-prettier": "^4.2.1",
```
Comment (NLmeng) [inline server/package.json:23]: these are needed, they are deleted, so there is not lint

File: server/index.js (hunk 1 above)
Comment (NLmeng) [reply server/index.js:32]: i think you deleted some necessary dependencies when fixing conflicts. if nothing is wrong, then when you commit, it should auto lint and format everything. You dont need to manually run check or format

**Pull request closed without merging**